jaraco/context.py

```diff
@@ -53,13 +53,35 @@ def tarball(
         runner('rm -Rf {target_dir}'.format(**vars()))
 
 
-@contextlib.contextmanager
-def tarball_cwd(*args, **kwargs) -> Iterator[str | os.PathLike]:
+def compose(*cmgrs):
     """
-    Convenience method for getting a tarball as the current working dir.
+    Compose any number of unary context managers into a single one.
+
+    Also allows the innermost context manager to take arbitrary arguments.
+
+    Like :func:`jaraco.functools.compose`, behavior works from right to
+    left, so the context manager should be indicated from outermost to
+    innermost.
```

Comment:
I'm a little uncertain about the order here. I'd like to be consistent, but I also think it's slightly unintuitive that the ordering is the reverse of how it's seen when chaining contexts in native syntax.

Reply:
I've decided to make the function private for now until I'm more confident about the ordering.